liquibase-standard/src/main/java/liquibase/analytics/LiquibaseAnalyticsListener.java

```diff
@@ -56,6 +61,38 @@ public int getPriority() {
 
     @Override
     public void handleEvent(Event event) throws Exception {
+        addSendEventsOnShutdownHook();
+        cachedEvents.add(event);
+        Integer maxCacheSize = Scope.getCurrentScope().get(Scope.Attr.maxAnalyticsCacheSize, getDefaultMaxAnalyticsCacheSize(event));
+        if (cachedEvents.size() >= maxCacheSize) {
+            flush();
+        } else {
+            Scope.getCurrentScope().getLog(getClass()).log(AnalyticsArgs.LOG_LEVEL.getCurrentValue(), "Caching analytics event to send later. Cache contains " + cachedEvents.size() + " event(s).", null);
+        }
+    }
+
+    private void addSendEventsOnShutdownHook() {
+        if (!addedShutdownHook.getAndSet(true)) {
+            Thread haltedHook = new Thread(() -> {
+                Scope.getCurrentScope().getLog(getClass()).fine("Sending " + cachedEvents.size() + " cached analytics events during shutdown hook");
+                try {
+                    flush();
+                } catch (Exception e) {
+                    Scope.getCurrentScope().getLog(getClass()).warning("Failed to send analytics events during shutdown hook.", e);
+                }
+            });
+            Runtime.getRuntime().addShutdownHook(haltedHook);
+        }
+    }
+
+    private int getDefaultMaxAnalyticsCacheSize(Event event) {
+        if (event != null && StringUtils.equals(event.getLiquibaseInterface(), Event.JAVA_API_INTEGRATION_NAME)) {
+            return 10;
```

Comment:
Just out of curiosity: 10 is a magic number or it was decided based on the statistics already reported?

Reply:
Totally arbitrary number.